### DISCUSSION TITLE
How to use NextAuth without database (only OAuth part)

### DISCUSSION THREAD

**Discussion by atumas-bananamilk** [nextauthjs/next-auth#5496] [Help]
### Question 💬

We use NextAuth to register / login users.

But now we're building a Facebook / Twitter **integration** (for people to able to tweet etc).
Since they all use OAuth2.0 why don't we reuse what NextAuth already has - the OAuth2.0 part?
But we don't want to store any of that information into "Account" table (which NextAuth does by default).
We just need **access / refresh tokens** ("Account" table is pretty limited).

We use Prisma + NextAuth for Social Login, but we want to re-use NextAuth's OAuth2.0 capabilities to get access tokens for our internal integrations.

I found this but no info on it:

![Screenshot 2022-10-02 at 15 59 05](https://user-images.githubusercontent.com/22228431/193455342-0d266466-e887-4ad3-a392-dac98f54bd83.png)

Can someone help me?

### How to reproduce ☕️

-

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by ThangHuuVu** [comment]
Hi, so just to be clear, by default NextAuth.js does not use DB if you don't specify an adapter. 🤔
> But we don't want to store any of that information into "Account" table (which NextAuth does by default).
We just need access / refresh tokens ("Account" table is pretty limited).

Can you explain a bit more here?

**Reply by atumas-bananamilk** [reply]
Thanks for replying.
We had to specify an adapter since we use Google for registrations to our platform. Our adapter is Prisma.

Now, we want to reuse NextAuth's OAuth2.0 part for Twitter / Facebook API (why build it ourselves?) because NextAuth gets the access / refresh tokens - exactly what we need to call Twitter / Facebook APIs.

This wouldn't be a problem if every platform user uses a unique Facebook account (I just call `signIn('facebook')` and info gets stored in "Account" table). Problems appear when I use the same Facebook account with another registered user (from "User" table), it doesn't get stored in "Account" table (some limitation between uniqueness of Account <-> User) - we don't want that.
We also have no control over that limitation.

Currently we avoided the limitation by deleting the automatically added "Account" row and recreating it in our own "SocialProfile" table on every `signIn` event:
![Screenshot 2022-10-06 at 00 54 27](https://user-images.githubusercontent.com/22228431/194171565-0836648a-b05f-4397-9e57-b99b8888daac.png)
![Screenshot 2022-10-06 at 00 55 38](https://user-images.githubusercontent.com/22228431/194171571-bd7dd151-a436-4fcc-83ec-1435829bc403.png)

Obviously this isn't the best approach but it works. If we could just re-use the NextAuth's OAuth2.0 part to get tokens WITHOUT storing info in "Account" table (only for Twitter / Facebook) it would be perfect as we only need access / refresh tokens.

**Reply by ThangHuuVu** [reply]
I believe you could do so by writing your own custom adapter. There's a guide here: https://next-auth.js.org/tutorials/creating-a-database-adapter. I think you could return nothing in `linkAccount` and it'll work 👀
Let me know if this helps!

**Reply by atumas-bananamilk** [reply]
Thanks, it works the way we implemented but will check `linkAccount` out!

**Reply by kumarabhirup** [reply]
@atumas-bananamilk did returning nothing in linkAccount work for you?

**Reply by atumas-bananamilk** [reply]
1 year later, we now have to add more providers and face the same problem.
We use Prisma adapter (https://next-auth.js.org/v3/adapters/prisma).
Is there a way to adapt it so we can define our own `linkAccount` within `PrismaAdapter`?
